packages/material-ui/src/Snackbar/Snackbar.js

```diff
@@ -130,37 +131,25 @@ const Snackbar = React.forwardRef(function Snackbar(props, ref) {
   const [exited, setExited] = React.useState(true);
 
   // Timer that controls delay before snackbar auto hides
-  const setAutoHideTimer = React.useCallback(
-    autoHideDurationParam => {
-      const autoHideDurationBefore =
-        autoHideDurationParam != null ? autoHideDurationParam : autoHideDuration;
-
-      if (!onClose || autoHideDurationBefore == null) {
-        return;
-      }
-
-      clearTimeout(timerAutoHide.current);
-      timerAutoHide.current = setTimeout(() => {
-        const autoHideDurationAfter =
-          autoHideDurationParam != null ? autoHideDurationParam : autoHideDuration;
-        if (!onClose || autoHideDurationAfter == null) {
-          return;
-        }
-        onClose(null, 'timeout');
-      }, autoHideDurationBefore);
-    },
-    [autoHideDuration, onClose],
-  );
+  const setAutoHideTimer = useEventCallback(autoHideDurationParam => {
```

Comment:
This uses a layout effect internally which is not necessary I believe. Looking at how useInterval is defined I think we can use useEffect or better extract this into a separate useTimeout hook so that it is crystal clear what is responsible for what.

https://overreacted.io/making-setinterval-declarative-with-react-hooks/#just-show-me-the-code